packages/twenty-server/src/engine/metadata-modules/object-metadata/utils/buildWorkspaceMigrationsForRemoteObject.ts

```diff
@@ -0,0 +1,181 @@
+  favoriteObjectMetadata: ObjectMetadataEntity,
+  schema: string,
+): WorkspaceMigrationTableAction[] => {
+  const createdObjectName = createdObjectMetadata.nameSingular;
+
+  return [
+    {
+      name: computeObjectTargetTable(activityTargetObjectMetadata),
+      action: 'alter',
+      columns: [
+        {
+          action: WorkspaceMigrationColumnActionType.CREATE,
+          columnName: `${computeCustomName(
+            createdObjectMetadata.nameSingular,
+            false,
+          )}Id`,
+          columnType: 'uuid',
+          isNullable: true,
+        } satisfies WorkspaceMigrationColumnCreate,
+      ],
+    },
+    {
+      name: computeObjectTargetTable(activityTargetObjectMetadata),
+      action: 'alter',
+      columns: [
+        {
+          action: WorkspaceMigrationColumnActionType.CREATE,
+          columnName: `${computeCustomName(
+            createdObjectMetadata.nameSingular,
+            false,
+          )}Id`,
+          columnType: 'uuid',
+        },
+      ],
+    },
+    {
+      name: computeObjectTargetTable(activityTargetObjectMetadata),
+      action: 'alter',
+      columns: [
+        {
+          action: WorkspaceMigrationColumnActionType.CREATE_COMMENT,
+          comment: `@graphql({"totalCount":{"enabled": true},"foreign_keys":[{"local_name":"${activityTargetObjectMetadata.nameSingular}Collection","local_columns":["${createdObjectName}Id"],"foreign_name":"${createdObjectName}","foreign_schema":"${schema}","foreign_table":"${createdObjectName}","foreign_columns":["id"]}]})`,
+        },
+      ],
+    },
+    // Add attachment relation
+    {
+      name: computeObjectTargetTable(attachmentObjectMetadata),
+      action: 'alter',
+      columns: [
+        {
+          action: WorkspaceMigrationColumnActionType.CREATE,
+          columnName: `${computeCustomName(
+            createdObjectMetadata.nameSingular,
+            false,
+          )}Id`,
+          columnType: 'uuid',
+          isNullable: true,
+        } satisfies WorkspaceMigrationColumnCreate,
+      ],
+    },
+    {
+      name: computeObjectTargetTable(attachmentObjectMetadata),
+      action: 'alter',
+      columns: [
+        {
+          action: WorkspaceMigrationColumnActionType.CREATE,
+          columnName: `${computeCustomName(
+            createdObjectMetadata.nameSingular,
+            false,
+          )}Id`,
+          columnType: 'uuid',
+        },
+      ],
+    },
+    {
+      name: computeObjectTargetTable(attachmentObjectMetadata),
+      action: 'alter',
+      columns: [
+        {
+          action: WorkspaceMigrationColumnActionType.CREATE_COMMENT,
+          comment: `@graphql({"totalCount":{"enabled": true},"foreign_keys":[{"local_name":"${attachmentObjectMetadata.nameSingular}Collection","local_columns":["${createdObjectName}Id"],"foreign_name":"${createdObjectName}","foreign_schema":"${schema}","foreign_table":"${createdObjectName}","foreign_columns":["id"]}]})`,
```

Comment:
Maybe this can be a util too?

Reply:
I extracted it into a separated function but in the same util since this is really specific :)